dotCMS/src/integration-test/java/com/dotcms/rest/api/v3/contenttype/FieldResourceTest.java

```diff
@@ -458,6 +459,137 @@ public void shouldMoveFields () throws DotSecurityException, DotDataException {
+     */
+    @Test
+    public void testGetRows_givenFieldsinJSON_ShouldDeserializeRespectingOrderAndDatatypes() throws DotSecurityException, DotDataException {
+
+        final ContentType type = createContentType();
+        final List<Map<String, Object>> fieldList = createTestFields (type.id());
+        final MoveFieldsForm form     = new MoveFieldsForm(fieldList);
+
+        final FieldLayout fieldLayout = form.getRows(type);
+
+        assertNotNull(fieldLayout);
+        assertNotNull(fieldLayout.getFields());
+        assertEquals(5, fieldLayout.getFields().size());
+        assertTrue(fieldLayout.getFields().get(0) instanceof RowField);
+        assertTrue(RowField.class.cast(fieldLayout.getFields().get(0)).dataType() == DataTypes.SYSTEM);
+        assertEquals("fields-0", RowField.class.cast(fieldLayout.getFields().get(0)).name());
+
+        assertTrue(fieldLayout.getFields().get(1) instanceof ColumnField);
+        assertTrue(ColumnField.class.cast(fieldLayout.getFields().get(1)).dataType() == DataTypes.SYSTEM);
+        assertEquals("fields-1", ColumnField.class.cast(fieldLayout.getFields().get(1)).name());
+
+        assertTrue(fieldLayout.getFields().get(2) instanceof TextField);
+        assertTrue(TextField.class.cast(fieldLayout.getFields().get(2)).dataType() == DataTypes.TEXT);
+        assertEquals("text", TextField.class.cast(fieldLayout.getFields().get(2)).name());
+
+        assertTrue(fieldLayout.getFields().get(3) instanceof TextField);
+        assertTrue(TextField.class.cast(fieldLayout.getFields().get(3)).dataType() == DataTypes.FLOAT);
+        assertEquals("decimal", TextField.class.cast(fieldLayout.getFields().get(3)).name());
+
+        assertTrue(fieldLayout.getFields().get(4) instanceof TextField);
+        assertTrue(TextField.class.cast(fieldLayout.getFields().get(4)).dataType() == DataTypes.INTEGER);
+        assertEquals("number", TextField.class.cast(fieldLayout.getFields().get(4)).name());
+
+    }
+
+    final List<Map<String, Object>> createTestFields (final String contentTypeId) {
+
+        final List<Map<String, Object>> fieldList = new ArrayList<>();
+        final Map<String, Object> rowField = CollectionsUtils.map("clazz" , "com.dotcms.contenttype.model.field.ImmutableRowField",
+                "contentTypeId", contentTypeId,
+                "dataType" , "SYSTEM",
+                "fieldContentTypeProperties", Collections.emptyList(),
+                "fieldType", "Row",
+                "fieldTypeLabel", "Row",
+                "fieldVariables", Collections.emptyList(),
+                "fixed", Boolean.FALSE,
+                "iDate", new Long(1584647620000l),
+                "id","a2d05a0b-5f4d-449d-9ca2-1117ab1ab086",
+                "indexed",  Boolean.FALSE,
+                "listed", Boolean.FALSE,
+                "modDate", new Long(1584647620000l),
+                "name", "fields-0",
+                "readOnly",Boolean.FALSE,
+                "required", Boolean.FALSE,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "required" appears 5 times in this file; the first occurrence is on line 522](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5214528)